File: test_src/r/first1000primes/001/primes.R
is_prime <- function(n) {
  if (n < 2) return(FALSE)
  if (n == 2) return(TRUE)
  if (n %% 2 == 0) return(FALSE)
  
  i <- 3
  while (i * i <= n) {
    if (n %% i == 0) return(FALSE)
    i <- i + 2
  }
  return(TRUE)
}

count <- 0
num <- 2
last_prime <- 0

while (count < 1000) {
  if (is_prime(num)) {
    last_prime <- num
    count <- count + 1
  }
  num <- num + 1
}

cat(last_prime, "\n", sep="")

File: test_src/python/first1000primes/001/primes.py
def count_primes():
    count = 0
    num = 2
    last_prime = 0
    
    while count < 1000:
        is_prime = True
        if num < 2:
            is_prime = False
        elif num == 2:
            is_prime = True
        elif num % 2 == 0:
            is_prime = False
        else:
            i = 3
            while i * i <= num:
                if num % i == 0:
                    is_prime = False
                    break
                i += 2
        
        if is_prime:
            last_prime = num
            count += 1
        num += 1
    
    print(last_prime)

count_primes()
